### PR TITLE
examples: expand and update

### PR DESCRIPTION
**Pull request by dignifiedquire**
## Runners
- [x] python
- [x] rust
- [ ] swift
- [ ] kotlin

###  python

- [x] author-create
- [x] author-list
- [x] ~~author-new~~
- [x] author-switch
- [x] blob-add
- [x] blob-list
- [x] blob-list-blobs
- [x] blob-list-collections
- [x] blob-list-incomplete-blobs
- [x] blob-share
- [x] blob-validate
- [x] doc-del
- [x] doc-delete
- [x] doc-drop
- [x] doc-export
- [x] doc-get
- [x] doc-import
- [x] doc-init
- [x] doc-join
- [x] doc-keys
- [x] doc-list
- [x] ~~doc-new~~
- [x] doc-create
- [x] doc-set
- [x] doc-share
- [x] doc-switch
- [x] doc-watch

### rust

- [ ] author-create
- [ ] author-list
- [ ] author-new
- [ ] author-switch
- [x] blob-add
- [ ] blob-list
- [ ] blob-list-blobs
- [ ] blob-list-collections
- [ ] blob-list-incomplete-blobs
- [ ] blob-share
- [ ] blob-validate
- [ ] doc-del
- [ ] doc-delete
- [ ] doc-drop
- [ ] doc-export
- [x] doc-get
- [ ] doc-import
- [ ] doc-init
- [ ] doc-join
- [ ] doc-keys
- [x] doc-list
- [ ] doc-new
- [x] doc-set
- [ ] doc-share
- [ ] doc-switch
- [ ] doc-watch
### swift

- [ ] author-create
- [ ] author-list
- [ ] author-new
- [ ] author-switch
- [ ] blob-add
- [ ] blob-list
- [ ] blob-list-blobs
- [ ] blob-list-collections
- [ ] blob-list-incomplete-blobs
- [ ] blob-share
- [ ] blob-validate
- [ ] doc-del
- [ ] doc-delete
- [ ] doc-drop
- [ ] doc-export
- [ ] doc-get
- [ ] doc-import
- [ ] doc-init
- [ ] doc-join
- [ ] doc-keys
- [ ] doc-list
- [ ] doc-new
- [ ] doc-set
- [ ] doc-share
- [ ] doc-switch
- [ ] doc-watch
### kotlin

- [ ] author-create
- [ ] author-list
- [ ] author-new
- [ ] author-switch
- [ ] blob-add
- [ ] blob-list
- [ ] blob-list-blobs
- [ ] blob-list-collections
- [ ] blob-list-incomplete-blobs
- [ ] blob-share
- [ ] blob-validate
- [ ] doc-del
- [ ] doc-delete
- [ ] doc-drop
- [ ] doc-export
- [ ] doc-get
- [ ] doc-import
- [ ] doc-init
- [ ] doc-join
- [ ] doc-keys
- [ ] doc-list
- [ ] doc-new
- [ ] doc-set
- [ ] doc-share
- [ ] doc-switch
- [ ] doc-watch

### CODE REVIEW

File: api-code-examples/python/doc-join.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-IROH_DATA_DIR = "./iroh_data_dir"
 
-# you'll need to get a ticket from somewhere, see the `doc share` command documentation
-# for details. This ticket will fail to join, but is a valid ticket.
-TICKET = "docaaa7qg6afc6zupqzfxmu5uuueaoei5zlye7a4ahhrfhvzjfrfewozgybl5kkl6u6fqcnjxvdkoihq3nbsqczxeulfsqvatb2qh3bwheoyahacitior2ha4z2f4xxk43fgewtcltemvzhaltjojxwqltomv2ho33snmxc6biajjeteswek4ambkabzpcfoajganyabbz2zplaaaaaaaaaagrjyvlqcjqdoaaioowl2ygi2likyov62rofk4asma3qacdtvs6whqsdbizopsefrrkx"
+async def main():
+    # Create in memory iroh node
+    node = await iroh.IrohNode.memory()
+    node_id = await node.node_id()
+    print("Started Iroh node: {}".format(node_id))
```
Comment (cclauss) [inline api-code-examples/python/doc-join.py:9]:
```suggestion
    print(f"Started Iroh node: {node_id}")
```

File: api-code-examples/python/doc-join.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-node = iroh.IrohNode(IROH_DATA_DIR)
-print("Started Iroh node: {}".format(node.node_id()))
+    # you'll need to get a ticket from somewhere, see the `doc share` command documentation
+    # for details. This ticket will fail to join, but is a valid ticket.
+    TICKET = "docaaa7qg6afc6zupqzfxmu5uuueaoei5zlye7a4ahhrfhvzjfrfewozgybl5kkl6u6fqcnjxvdkoihq3nbsqczxeulfsqvatb2qh3bwheoyahacitior2ha4z2f4xxk43fgewtcltemvzhaltjojxwqltomv2ho33snmxc6biajjeteswek4ambkabzpcfoajganyabbz2zplaaaaaaaaaagrjyvlqcjqdoaaioowl2ygi2likyov62rofk4asma3qacdtvs6whqsdbizopsefrrkx"
 
-doc = node.doc_join(TICKET)
-print("Joined doc: {}".format(doc.id()))
+    doc = await node.doc_join(TICKET)
+    print("Joined doc: {}".format(doc.id()))
```
Comment (cclauss) [inline api-code-examples/python/doc-join.py:16]:
```suggestion
    print(f"Joined doc: {doc.id()}")
```

File: api-code-examples/python/doc-keys.py [hunk 3]
```diff
@@ -1,34 +1,39 @@
...
-node = iroh.IrohNode(IROH_DATA_DIR)
-print("Started Iroh node: {}".format(node.node_id()))
+async def main():
+    # Create in memory iroh node
+    node = await iroh.IrohNode.memory()
 
-author = node.author_create()
-print("Created author: {}".format(author.to_string()))
+    node_id = await node.node_id()
+    print("Started Iroh node: {}".format(node_id))
```
Comment (cclauss) [inline api-code-examples/python/doc-keys.py:10]:
```suggestion
    print(f"Started Iroh node: {node_id}")
```

File: api-code-examples/python/doc-keys.py [hunk 4]
```diff
@@ -1,34 +1,39 @@
...
 
-doc = node.doc_create()
-print("Created doc: {}".format(doc.id()))
+    author = await node.author_default()
 
-for i, key in enumerate(['a', 'b', 'c']):
-    doc.set_bytes(author, bytes(key, "utf8"), bytes(str(i), "utf8"))
+    doc = await node.doc_create()
+    print("Created doc: {}".format(doc.id()))
```
Comment (cclauss) [inline api-code-examples/python/doc-keys.py:15]:
```suggestion
    print(f"Created doc: {doc.id()}")
```

File: api-code-examples/python/doc-keys.py [hunk 5]
```diff
@@ -1,34 +1,39 @@
...
+
+    # get all the entries with default filtering and sorting
+    query = iroh.Query.all(None)
+    entries = await doc.get_many(query)
+    print("Keys:")
+    for entry in entries:
+        key = entry.key()
+        hash = entry.content_hash()
+        content = await entry.content_bytes(doc)
+        print("{} : {} (hash: {})".format(key, content.decode("utf8"), hash))
```
Comment (cclauss) [inline api-code-examples/python/doc-keys.py:28]:
```suggestion
        print(f"{key} : {content.decode('utf8')} (hash: {hash})")
```

File: api-code-examples/python/doc-list.py [hunk 6]
```diff
@@ -1,24 +1,26 @@
 import iroh
+import asyncio
 
-IROH_DATA_DIR = "./iroh_data_dir"
+async def main():
+    # Create in memory iroh node
+    node = await iroh.IrohNode.memory()
+    node_id = await node.node_id()
+    print("Started Iroh node: {}".format(node_id))
```
Comment (cclauss) [inline api-code-examples/python/doc-list.py:8]:
```suggestion
    print(f"Started Iroh node: {node_id}")
```

File: api-code-examples/python/doc-list.py [hunk 7]
```diff
@@ -1,24 +1,26 @@
...
 
-node = iroh.IrohNode(IROH_DATA_DIR)
-print("Started Iroh node: {}".format(node.node_id()))
+    # create a document
+    doc = await node.doc_create()
+    print("Created doc: {}".format(doc.id()))
```
Comment (cclauss) [inline api-code-examples/python/doc-list.py:12]:
```suggestion
    print(f"Created doc: {doc.id()}")
```

File: api-code-examples/python/doc-list.py [hunk 8]
```diff
@@ -1,24 +1,26 @@
...
 
-# create a document
-doc = node.doc_create()
-print("Created doc: {}".format(doc.id()))
+    # create a second document
+    doc = await node.doc_create()
+    print("Created doc: {}".format(doc.id()))
```
Comment (cclauss) [inline api-code-examples/python/doc-list.py:16]:
```suggestion
    print(f"Created doc: {doc.id()}")
```

File: api-code-examples/python/doc-list.py [hunk 9]
```diff
@@ -1,24 +1,26 @@
...
-print("List all {} docs:".format(len(docs)))
-# doc ids are also called "namespace ids"
-for namespace_and_capability in docs:
-    print("\t{}".format(namespace_and_capability.namespace))
+    # list all your documents
+    docs = await node.doc_list();
+    print("List all {} docs:".format(len(docs)))
+    # doc ids are also called "namespace ids"
+    for namespace_and_capability in docs:
+        print("\t{}".format(namespace_and_capability.namespace))
```
Comment (cclauss) [inline api-code-examples/python/doc-list.py:23]:
```suggestion
    print(f"List all {len(docs)} docs:")
    # doc ids are also called "namespace ids"
    for namespace_and_capability in docs:
        print(f"\t{len(docs)}")
```

File: api-code-examples/python/author-list.py [hunk 10]
```diff
@@ -1,20 +1,25 @@
...
+import asyncio
 
-IROH_DATA_DIR = "./iroh_data_dir"
 
-node = iroh.IrohNode(IROH_DATA_DIR)
-print("Started Iroh node: {}".format(node.node_id()))
+async def main():
+    node = await iroh.IrohNode.memory()
+    node_id = await node.node_id()
+    print("Started Iroh node: {}".format(node_id))
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply api-code-examples/python/author-list.py]: thanks a lot, used `ruff` to cleanup all the python examples

**Review by b5 — APPROVED** [review]
damn this is a lot of work. thank you!

**Review by cclauss — APPROVED** [review]

**Pull request merged**